packages/components/src/components/lifecycleValidation.spec.ts

```diff
@@ -182,6 +186,23 @@ it.each<TagName>(tagNamesWithSlottedCss)(
   }
 );
 
+it.each<TagName>(tagNamesWithSlotAndTheme)(
+  'should call getDataThemeDarkAttribute() with correct parameters via render for %s',
```

Comment:
The test indeed does what the description here promises. But that doesn't guarantee that it is used correctly.
Better and safer would be to check that the attribute is set on the host element. Currently the function could just be called anywhere.

Reply:
as we discussed, I added an e2e test for this scenario to default-dom.e2e.ts